code/workspaces/infrastructure-api/src/dataaccess/testUtil/databaseMock.js

```diff
@@ -1,17 +1,33 @@
-function createDatabaseMock(items, state = {}) {
-  let lastInvocation = state;
+class InvocationState {
+  constructor() {
+    this.queries = [];
+    this.lastQuery = undefined;
+    this.entity = undefined;
+    this.projectKey = undefined;
+    this.params = undefined;
+    this.user = undefined;
+  }
+
+  addQuery(query) {
+    this.lastQuery = query;
+    this.queries.push(query);
+  }
+}
+
+function createDatabaseMock(items, state) {
+  let lastInvocation = state || new InvocationState();
 
   return () => ({
     find: (query) => {
-      lastInvocation.query = query;
+      lastInvocation.addQuery(query);
       return createDatabaseMock(items, lastInvocation)();
     },
     findOne: (query) => {
-      lastInvocation.query = query;
+      lastInvocation.addQuery(query);
```

Comment:
Should we rename this to something like `invocations`?

Reply:
That sounds like a good idea